sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/survnet/SurvNetSteps.java

```diff
@@ -130,6 +131,102 @@ public SurvNetSteps(WebDriverHelpers webDriverHelpers, SoftAssert softly) {
           softly.assertAll();
         });
 
+    And(
+        "I check if Reinfection option is set in SORMAS generated single XML file is correct",
+        () -> {
+          String reinfectionOptionFromUI = survnetCase.getReinfection();
+          String expectedReinfectionOption;
+
+          if (reinfectionOptionFromUI.equals("JA")) expectedReinfectionOption = "true";
+          else expectedReinfectionOption = "false";
+
+          softly.assertEquals(
+              getValueFromSpecificFieldByName(singleXmlFile, "Reinfection"),
+              expectedReinfectionOption,
+              "Reinfection option is not set!");
+          softly.assertAll();
+        });
+
+    And(
+        "I check that LabInfoAvailable is change in SORMAS generated single XML file is correct",
+        () -> {
+          softly.assertEquals(
+              getValueFromSpecificFieldByName(singleXmlFile, "LabInfoAvailable"),
+              "20",
+              "LabInfoAvailable has incorrect value");
+          softly.assertAll();
+        });
+
+    And(
+        "I check if Reinfection \"([^\"]*)\" checkbox value in SORMAS generated single XML file is correct",
+        (String reinfectionCheckbox) -> {
+          switch (reinfectionCheckbox) {
+            case "PREVIOUS COVID INFECTION IS KNOWN":
+              softly.assertEquals(
+                  getValueFromSpecificFieldByName(singleXmlFile, "ReinfectionDetailCVD01"),
+                  PreviousCovidInfectionIsKnownValue.toString(),
+                  "checkbox value in xml is incorrect");
```

Comment:
It would be helpful if you add info in a message which exacly checkbox value in the xml is incorrect for example "Previous covid infection checkbox in xml file is incorrect"